R-package/R/xgb.DMatrix.R

```diff
@@ -58,19 +77,35 @@ xgb.DMatrix <- function(
   qid = NULL,
   label_lower_bound = NULL,
   label_upper_bound = NULL,
-  feature_weights = NULL
+  feature_weights = NULL,
+  as_quantile_dmatrix = FALSE,
+  ref = NULL,
+  max_bin = NULL
```

Comment:
It's probably better to split it into a sub class.

Reply:
Yes, since the code paths will be entirely different, better to have a dedicated function `xgb.QuantileDMatrix` like in the python interface.